### DISCUSSION TITLE
Auto generated OpenAPI documentation

### DISCUSSION THREAD

**Discussion by Butch78** [shuttle-hq/shuttle#323] [Ideas]
It would be cool for Shuttle to have to Swagger out of the box, I think it would make it much easier for new users. 

It one of the best features of https://fastapi.tiangolo.com/ and I think It would really improve the user experience

**Comment by Mouwrice** [comment]
You can achieve this by using poem openapi https://github.com/poem-web/poem/blob/master/poem-openapi/README.md

It is a framework that ensures your entire web api is conformant with OpenAPI and integrates with swagger, openapi-explorer and other tools.
